### PR TITLE
Allow standalone foreman::repo usage and add acceptance testing

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: spec/acceptance/foreman_cli_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper_acceptance'
+
+describe 'Scenario: install foreman-cli without foreman' do
+  before(:context) do
+    case fact('osfamily')
+    when 'RedHat'
+      on default, 'yum -y remove foreman* && rm -rf /etc/yum.repos.d/foreman*.repo'
```
Comment (domcleal) [inline spec/acceptance/foreman_cli_spec.rb]: Are there other packages, such as `tfm-*` (or more specifically, `tfm-rubygem-hammer_cli_*`) that should be removed too, to ensure a consistent state?

File: spec/acceptance/foreman_cli_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper_acceptance'
+
+describe 'Scenario: install foreman-cli without foreman' do
+  before(:context) do
```
Comment (domcleal) [inline spec/acceptance/foreman_cli_spec.rb]: Should cleanup be _after_?

File: spec/acceptance/foreman_cli_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    case fact('osfamily')
+    when 'RedHat'
+      on default, 'yum -y remove foreman* && rm -rf /etc/yum.repos.d/foreman*.repo'
+    when 'Debian'
+      on default, 'apt-get purge foreman* && rm -rf /etc/apt/sources.list.dforeman*'
```
Comment (domcleal) [inline spec/acceptance/foreman_cli_spec.rb]: 1. Missing `/` after `sources.list.d`
2. Ditto, `ruby-hammer-cli-*`?

**Comment by mmoll** [comment]
also, this needs a rebase now

File: spec/acceptance/foreman_cli_spec.rb (hunk 2 above)
Comment (ekohl) [inline spec/acceptance/foreman_cli_spec.rb]: My thinking was that this ensures a clean test. Beaker can reuse the container for multiple tests and adding cleanup in every test is likely to miss something.

**Comment by ekohl** [comment]
Updated

**Comment by domcleal** [comment]
Merged, thanks @ekohl.

**Pull request merged**